server/api.js

```diff
@@ -55,18 +55,37 @@ router.post("/bookings", async function (req, res) {
 			"SELECT id from desk_user WHERE username=$1",
 			[userName]
 		);
-		// const tableResult = await db.query(
-		// 	"SELECT id from desk WHERE desk_table_name=$2",
-		// 	[deskName]
-		// );
+
 		if (userResult.rows.length === 0) {
 			return res.status(400).send({
 				message: `User with name ${userName} does not exist`,
 				field: "name",
 			});
 		}
+
+		const bookingsByDayResult = await db.query(
+			"select * from booking where booking_date = $1",
+			[bookingDate]
+		);
+
+		if (bookingsByDayResult.rows.length >= 5) {
+			return res.status(400).send({
+				message: `No desks available for ${bookingDate}`,
+				field: "date",
+			});
+		}
+
+		if (deskId) {
```

Comment:
instead of if(DeskId) maybe put if (deskId !== undefined) maybe deskId = 0 is a real deskId